modules/wallet/update.go

```diff
@@ -10,21 +10,93 @@ import (
 	"github.com/NebulousLabs/bolt"
 )
 
-// historicOutput defines a historic output as recognized by the wallet. This
-// struct is primarily used to sort the historic outputs before inserting them
-// into the bolt database.
-type historicOutput struct {
-	id  types.OutputID
-	val types.Currency
+// threadedResetSubscriptions unsubscribes the wallet from the consensus set and transaction pool
+// and subscribes again.
+func (w *Wallet) threadedResetSubscriptions() error {
+	if !w.scanLock.TryLock() {
```

Comment:
I guess that we don't strictly need to interact with the threadgroup here, actually. Grabbing an `Add` on the threadgroup would stall shutdown until the whole rescan finishes, which is not what we want I think. I'm not sure actually how to get a clean interrupt on the subscribe function, but it seems like something that we would want.